src/basicTypes.ts

```diff
@@ -0,0 +1,125 @@
+import { Vector3, Material } from 'three';
+
+export type EasingFunction = (x: number) => number;
+
+export type HeightmapFunction = ((g: Vector3[], options: TerrainOptions) => void);
+
+export interface TerrainOptions {
```

Comment:
I guess I will need you help, here. I wasn't able to always figure out if the options were needed or not. Sometimes, when I used too much Pick or Partial, I ran into a dead end because I could not make it work. Most of the places where TerrainOptions is used are called from Terrain() anyway so that the options all have default values.

Can you help here, please?

Reply:
If you change `export interface TerrainOptions {` to `export type TerrainOptions = Partial<{` and recompile, you will get 164 errors, each listing a property that should be available. For example, `GaussianBoxBlur` will complain that it's missing `options.xSegments` and `options.ySegments`.

I acknowledge that this is a substantial update. As a compromise, how about we just make the most commonly used properties required?

```ts
export interface TerrainOptions {
    after?: ((v: Vector3[], options: TerrainOptions) => void) | null;
    easing?: EasingFunction;
    heightmap?: HTMLCanvasElement | ImageBitmap | HeightmapFunction;
    material?: Material | null;
    maxHeight: number;
    minHeight: number;
    optimization?: Optimization;
    frequency: number;
    steps?: number;
    stretch?: boolean;
    turbulent?: boolean;
    useBufferGeometry?: boolean;
    xSegments: number;
    xSize: number;
    ySegments: number;
    ySize: number;
}
```